saleor/payment/gateways/braintree/__init__.py

```diff
@@ -136,101 +139,111 @@ def authorize(payment_information: Dict, connection_params: Dict) -> Dict:
-            'transaction_id', payment_information['token']),
-        'error': error,
-        'raw_response': gateway_response}
-
-
-def refund(payment_information: Dict, connection_params: Dict) -> Dict:
+    return GatewayResponse(
+        is_success=result.is_success,
+        kind=TransactionKind.VOID,
+        amount=gateway_response.get(
+            'amount', payment_information.amount),
+        currency=gateway_response.get(
+            'currency', payment_information.currency),
+        transaction_id=gateway_response.get(
+            'transaction_id', payment_information.token),
+        error=error,
+        raw_response=gateway_response
+    )
+
+
+def refund(
+        payment_information: Dict, connection_params: Dict) -> GatewayResponse:
     gateway = get_braintree_gateway(**connection_params)
 
     try:
         result = gateway.transaction.refund(
-            transaction_id=payment_information['token'],
-            amount_or_options=str(payment_information['amount']))
+            transaction_id=payment_information.token,
+            amount_or_options=str(payment_information.amount))
     except braintree_sdk.exceptions.NotFoundError:
         raise BraintreeException(DEFAULT_ERROR_MESSAGE)
 
     gateway_response = extract_gateway_response(result)
     error = get_error_for_client(gateway_response['errors'])
 
-    return {
-        'is_success': result.is_success,
-        'kind': TransactionKind.REFUND,
-        'amount': gateway_response.get(
-            'amount', payment_information['amount']),
-        'currency': gateway_response.get(
-            'currency', payment_information['currency']),
-        'transaction_id': gateway_response.get(
-            'transaction_id', payment_information['token']),
-        'error': error,
-        'raw_response': gateway_response}
+    return GatewayResponse(
+        is_success=result.is_success,
+        kind=TransactionKind.REFUND,
+        amount=gateway_response.get(
+            'amount', payment_information.amount),
+        currency=gateway_response.get(
+            'currency', payment_information.currency),
+        transaction_id=gateway_response.get(
+            'transaction_id', payment_information.token),
+        error=error,
+        raw_response=gateway_response
+
+    )
 
 
 def process_payment(
-        payment_information: Dict, connection_params: Dict) -> Dict:
+        payment_information: PaymentInformation, connection_params: Dict
+) -> GatewayResponse:
     auth_resp = authorize(payment_information, connection_params)
-    if auth_resp['is_success']:
-        payment_information['token'] = auth_resp['transaction_id']
-        return [auth_resp, capture(payment_information, connection_params)]
-    return [auth_resp, void(payment_information, connection_params)]
+    if auth_resp.is_success:
+        payment_information.token = auth_resp.transaction_id
+        return capture(payment_information, connection_params)
+    return auth_resp
```

Comment:
We have to update our docs since the API has changed and now it returns a single object instead of a list: https://docs.getsaleor.com/en/latest/guides/payments.html?highlight=process_payment#process-payment-payment-information-connection-params